Add-ons/UmbracoForms/Developer/Extending/Adding-a-Workflowtype.md

```diff
@@ -50,6 +50,20 @@ namespace MyFormsExtensions
 
                 // or get it as a string
                 rf.ValuesAsString();
+
+                // set or replace values against a field in the current form submission
+                if(rf.Alias.Equals("myField", StringComparison.InvariantCulture))
+                {
+                    rf.Values.Clear();
+                    rf.Values.Add("My new value");
+                }
+
+                // Update the form submission via the UpdateRecord method, but only when the Workflow is in an Approval chain. If not in an Approval chain, the values are still saved against the RecordField.
```

Comment:
This though I'm not so sure about - i.e. updating a record within a loop that iterates through the fields.  Maybe it works OK in the context it was used in, but we have to be 100% sure it's a good idea in the general case before using it in the docs.  So I think we should remove this part, or at least move it outside of the loop.

Reply:
To be honest, I only added the example in the context of the already existing snippet / loop. On my own project, I've actually done it slightly differently (outside of a loop). 

```
RecordField? myField = context.Record.RecordFields.Values.FirstOrDefault(x => x.Alias.Equals("myFieldAlias", StringComparison.InvariantCulture)) ?? throw new Exception($"Could not find field 'myFieldAlias'");
myField.Values.Clear();
myField.Values.Add("My new value");
```

This could sit either side of the loop if you're happy with it?

I personally feel this information would be super useful to anybody needing to perform a similar task inside their own custom workflows.

Reply:
It sounds like we should try to put the code outside the loop here.
Would you be up for making the changes @AndyBoot ?